examples/grpc/src/main/java/io/opentelemetry/example/HelloWorldClient.java

```diff
@@ -0,0 +1,163 @@
+import io.opentelemetry.sdk.trace.SpanData;
+import io.opentelemetry.sdk.trace.TracerSdkFactory;
+import io.opentelemetry.sdk.trace.export.SimpleSpansProcessor;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.Tracer;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class HelloWorldClient {
+  private static final Logger logger = Logger.getLogger(HelloWorldClient.class.getName());
+  private final ManagedChannel channel;
+  private final String hostname;
+  private final Integer port;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  // OTel API
+  Tracer tracer;
+  // Export traces in memory
+  InMemorySpanExporter inMemexporter = InMemorySpanExporter.create();
+  // Share context via text headers
+  HttpTextFormat<SpanContext> textFormat;
+  // Inject context into the gRPC request metadata
+  HttpTextFormat.Setter<Metadata> setter =
+      new HttpTextFormat.Setter<Metadata>() {
+        @Override
+        public void put(Metadata carrier, String key, String value) {
+          carrier.put(Metadata.Key.of(key, Metadata.ASCII_STRING_MARSHALLER), value);
+        }
+      };
+
+  /** Construct client connecting to HelloWorld server at {@code host:port}. */
+  public HelloWorldClient(String host, int port) {
+    this.hostname = host;
+    this.port = port;
+    this.channel =
+        ManagedChannelBuilder.forAddress(host, port)
+            // Channels are secure by default (via SSL/TLS). For the example we disable TLS to avoid
+            // needing certificates.
+            .usePlaintext()
+            // Intercept the request to tag the span context
+            .intercept(new OpenTelemetryClientInterceptor())
+            .build();
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
+    // Initialize the OTel tracer
+    initTracer();
+  }
+
+  private void initTracer() {
+    // Get the tracer
+    TracerSdkFactory tracerFactory = OpenTelemetrySdk.getTracerFactory();
+    // Set to process the spans in memory
+    tracerFactory.addSpanProcessor(SimpleSpansProcessor.newBuilder(inMemexporter).build());
+    // Give a name to the tracer
+    this.tracer = tracerFactory.get("io.opentelemetry.example.HelloWorldClient");
+    textFormat = this.tracer.getHttpTextFormat();
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /** Say hello to server. */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+
+    // Start a span
+    Span span =
+        tracer.spanBuilder("helloworld.Greeter/SayHello").setSpanKind(Span.Kind.CLIENT).startSpan();
+    // TODO provide attributes to Span.Builder
+    span.setAttribute("component", "grpc");
+    span.setAttribute("rpc.service", "helloworld.Greeter");
```

Comment:
```suggestion
    span.setAttribute("rpc.service", "Greeter");
```
see https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-rpc.md#attributes